File: 프로그래머스/unrated/120902. 문자열 계산하기/문자열 계산하기.py
def solution(my_string):
    string = my_string.split(" ")
    answer = int(string[0])
    i=1
    while i<len(string):
        if string[i] == '+':
            answer += int(string[i+1])
            i+=2
        elif string[i] == '-':
            answer -= int(string[i+1])
            i+=2
            
    return answer

File: 프로그래머스/0/120902. 문자열 계산하기/문자열 계산하기.java
class Solution {
    public int solution(String my_string) {
        int answer = 0;
        String[] str = my_string.split(" ");
        
        answer = Integer.parseInt(str[0]);
        boolean plus = true;
        for(int i=1;i<str.length;i++){
            if(i%2 == 0){
                if(plus){
                    answer += Integer.parseInt(str[i]);
                }
                else{
                    answer -= Integer.parseInt(str[i]);
                }
            }
            else{
                if(str[i].equals("+")){
                    plus = true;
                }
                else{
                    plus = false;
                }
            }
        }
        
        return answer;
    }
}